Add "Introduction to DApps" concept (#962)
casper-network/docs: source/docs/casper/concepts/intro-to-dapps.md

---
title: Introduction to DApps
slug: /intro-to-dapps
---

> # Introduction to DApps (Distributed Applications)

Reviewer: Yeah, I mean the idea was to capitalize the important nouns in the title, in the way that titles often do. I guess the visual impact is similar when the A is capitalized.

- Isolated: DApps are executed in an isolated situation. If a DApp has a bug, it won't affect the normal functioning of the Blockchain network.

So a DApp is not just a client-server application where the application can do some work offline, nor a web application which can operate in a disconnected mode, where there is a central server which acts as a repository of information and business logic. A DApp is conceived and built using a distributed architecture where the network of nodes does the processing of smart contracts instead of a central server.

## Benefits of a DApp

1. Durability/uptime characteristics. Because of the fact that each operation on a DApp is processed by one or many identical nodes in a network, if one node goes down it has no effect on the overall operation of the network. DApps can maintain 24/7, uninterrupted uptime without investment in massive data centers or secure facilities. A DApp cannot be attacked by Denial of Service (DDOS) attacks.
2. User Privacy: You do not need to provide some real-world identification, such as a GMail or Facebook account, to interact with a Blockchain network.

> 3. Resistance to outside control. No single entity can block another from accessing the blockchain, reading data, deploying or interacting with DApps, or submitting transactions to the chain.

Reviewer 2: ```suggestion
3. Resistance to outside control

No single entity can block another from accessing the blockchain, reading data, deploying or interacting with dApps, or submitting transactions to the chain.
```

> 4. Trust in the state of the block chain: The data stored on the blockchain is immutable and indisputable, rather than relying on a centralized authority. Users may not change the transaction record or other data that has already been launched to the public.

Reviewer 2: ```suggestion
4. Trust in the state of the blockchain

Data stored on the blockchain is immutable and decentralized, rather than relying on a centralized authority. Users may not change the transaction record or any other data present on the blockchain.
```

> ## Downsides to DApps

Reviewer 2: ```suggestion
## Downsides to dApps
```

> 1. Maintenance: Compared to traditional applications, once a DApp is published, it is more difficult to maintain as the data is stored on the Blockchain which is immutable. If bugs are found, developers can’t fix them as easily as with a traditional client-server architecture.

Reviewer 2: ```suggestion
1. Maintenance

Compared to traditional applications, once a DApp is published, it is more difficult to maintain. The data is stored immutably on the blockchain. If bugs are found, developers can’t fix them as easily as with a traditional client-server architecture.
```

> 2. Network Efficiency: DApp networks and platforms need to agree consensus before mutating the state of the chain. This means that processing millions of transactions per second, such as in a low-latency trading or betting application, is not possible. If blistering network performance is a prerequisite, then perhaps DApps are not the correct architecture for the application.

Reviewer 2: ```suggestion
2. Network Efficiency

DApp networks and platforms need to reach consensus before mutating the state of the chain. This means that processing millions of transactions per second, such as in a low-latency trading or betting applications, is not possible. Dapps are not the correct architecture for applications that require high network performance.
```

Reviewer: Why add a line here, but not for the previous paragraph?

## Some popular use cases

> 1. DeFi.

Reviewer 2: ```suggestion
1. DeFi
```

>   [De-Fi (Decentralized Finance)](https://www.investopedia.com/decentralized-finance-defi-5113835) refers to a movement towards financial systems without the need for centralized authorities, like banks or governments, to act as intermediaries and verify transactions on them. This is enabled by the "trustless" model provided under a blockchain system, where the validity of a transaction is ensured by the very nature of the platform, not by any over-arching organization. This represents a challenge to traditional systems of digitized finance, with additional possibilities and concerns.

Reviewer 2: ```suggestion
[De-Fi (Decentralized Finance)](https://www.investopedia.com/decentralized-finance-defi-5113835) refers to a movement towards financial systems without the need for centralized authorities, like banks or governments, to act as intermediaries and verify transactions. This is enabled by the "trustless" model provided under a blockchain system, where the validity of a transaction is ensured by the very nature of the platform. This represents a challenge to traditional systems of digital finance, with additional possibilities and concerns.
```

> 2. Gaming apps and NFTs.

Reviewer 2: ```suggestion
2. Gaming apps and NFTs
```

>   [Game-Fi](https://gamefi.org/) is a gaming model based on blockchain tech. The idea is that users of a gaming app can earn crypto and non-fungible tokens [(NFT)](https://www.investopedia.com/non-fungible-tokens-nft-5115211) which can then be traded in the market, so they can securely earn money while gaming.

Reviewer 2: ```suggestion
  [Game-Fi](https://gamefi.org/) is a gaming model based on blockchain technology. Through GameFi, users of a gaming app can earn cryptocurrency and non-fungible tokens [(NFT)](/concepts/glossary/N/#non-fungible-token). These items can then be traded in the market, allowing players to securely earn money while gaming.
```

---
title: Introduction to dApps
---

# Introduction to dApps (Distributed Applications) on the Casper network

### What is a dApp?

DApp stands for **Decentralized Application**. Specifically, it refers to an application built on a blockchain network which combines smart contracts and a user interface. 
To be considered a dApp, an application should fulfill the following requirements:

#### Characteristics of a dApp

> - Decentralized: DApps are run by thousands of nodes around the globe. No party can take control of it, including governments and the developers.

Reviewer: Yes, I mean... the intent of the sentence is other than that, Casper customers of course can affect their own networks. i feel like this section is just confusing and feels a little out of place. it would be better to link to an external definition or information page for those who literally don't understand the point of a dApp

- Deterministic: The execution of DApps is not affected by any changes in the environment.

> - Turing complete: DApps can perform any action given the required resources.

Reviewer 2: Casper is psuedo-Turing Complete, so we should not claim that Turing Complete is a requirement for a dApp.

- Isolated: DApps are executed in an isolated situation. If a DApp has a bug, it won't affect the normal functioning of the Blockchain network.

A decentralized system consists of a group of interchangeable machines that can perform as a full system or distributed database. Additional machines strengthen the overall system by adding redundancy and computational power.

A dApp is not just a client-server application where the application can do some work offline, nor is it a web application which can operate in a disconnected mode. A dApp is conceived and built using a distributed architecture where a network of [nodes](../concepts/glossary/N.md#node) does the processing of smart contracts instead of a single central server.

Any dApp will need access to a decentralized network, in one form or another. In a Casper network, this means connecting to a [node](../concepts/glossary/N.md#node). The decentralized aspect creates a situation where each node is fundamentally interchangeable for this purpose. If the connected node fails, the dApp can switch to a different node and continue operating without losing data or functionality.

> #### Some use cases for dApps

Reviewer 2: ```suggestion
#### Some use cases for dApps
```

I feel like this section isn't really "documentation". Use cases fall more within a blog or article.
